reconcile/terraform_resources.py

```diff
@@ -533,49 +533,52 @@ def runner(
 
 def early_exit_desired_state(*args: Any, **kwargs: Any) -> dict[str, Any]:
     gqlapi = gql.get_api()
-    namespaces = get_tf_namespaces()
-    resources = []
-    for ns_info in namespaces:
+    state_for_accounts = {
+        account["name"]: {
+            "meta": account,
+            "specs": {},
+        }
+        for account in queries.get_aws_accounts(terraform_state=True)
+    }
+    for ns_info in get_tf_namespaces():
         for spec in get_external_resource_specs(
             ns_info.dict(by_alias=True), provision_provider=PROVIDER_AWS
         ):
+            spec_resources: dict[str, str] = {}
+            spec_state = {
+                "spec": asdict(spec),
+                "resources": spec_resources,
+            }
 
-            def register_resource(
-                spec: ExternalResourceSpec, resource: Optional[dict[str, Any]]
-            ) -> None:
+            def register_resource(resource: Optional[dict[str, Any]]) -> None:
                 if resource:
-                    resources.append({
-                        IDENTIFIER_FIELD_NAME: f"{spec.cluster_name}/{spec.namespace_name}/{spec.provisioner_name}/{spec.provider}/{spec.identifier}/{resource.get('path')}",
-                        "content_sha": resource.get("sha256sum"),
-                        "provisioner": spec.provisioner_name,
-                    })
+                    spec_resources[resource["path"]] = resource["sha256sum"]
 
             defaults = spec.resource.get("defaults")
             if defaults:
-                register_resource(spec, gqlapi.get_resource(defaults))
+                register_resource(gqlapi.get_resource(defaults))
             parameter_group = spec.resource.get("parameter_group")
             if parameter_group:
-                register_resource(spec, gqlapi.get_resource(parameter_group))
+                register_resource(gqlapi.get_resource(parameter_group))
             for spec_item in spec.resource.get("specs") or []:
                 defaults = spec_item.get("defaults")
                 if defaults:
-                    register_resource(spec, gqlapi.get_resource(defaults))
+                    register_resource(gqlapi.get_resource(defaults))
+            spec_id = f"{spec.cluster_name}/{spec.namespace_name}/{spec.provisioner_name}/{spec.provider}/{spec.identifier}"
+            state_for_accounts[spec.provisioner_name][spec_id] = spec_state
```

Comment:
maybe need to add a test here, I think you mean to populate `state_for_accounts[spec.provisioner_name]["specs"][spec_id] = spec_state`, here extract function and use in comprehension would make it cleaner too.

Reply:
fixed by your comprehension proposal

Reply:
@geoberle it's not fixed, `specs` field in `state_for_accounts` is always empty, not a big issue as no conflict between `spec_id` and `meta`. Either remove `specs` from `state_for_accounts` or put specs under `specs`.